data_model/src/block.rs

```diff
@@ -399,13 +399,15 @@ pub mod stream {
 
     #[model]
     pub mod model {
+        use core::num::NonZeroU64;
+
         use super::*;
 
         /// Request sent to subscribe to blocks stream starting from the given height.
         #[version_with_scale(n = 1, versioned = "VersionedBlockSubscriptionRequest")]
         #[derive(Debug, Clone, Copy, Constructor, Decode, Encode, IntoSchema)]
         #[repr(transparent)]
-        pub struct BlockSubscriptionRequest(pub u64);
+        pub struct BlockSubscriptionRequest(pub NonZeroU64);
```

Comment:
Such a change should be reflected in the generated schema, so that SDKs have a way to handle it specifically.

Also, such a type is a part of the "magic types" list: #3535

Reply:
> Such a change should be reflected in the generated schema, so that SDKs have a way to handle it specifically.

I would change the phrasing to `could` instead of `should`.

I'm not sure I entirely agree here. There is a point to be made about simplicity. Do we really want to overload schema with handling niche values for types? If sdk devs want to they still can do so even if it is not specified in the schema. Some implicit knowledge might be fine

On the other hand do we want to consider introducing some sort of type metadata in the schema? This info could then be embedded in the schema's metadata

Reply:
So rather than having a distinct type for every non-robust version of the container type we could rather decorate the type with metadata

Reply:
I take back what I've said. Look [here](https://github.com/hyperledger/iroha/issues/3535#issuecomment-1579905464). All information has to be in the type name, we can't use type metadata. So if we want this we need a special type

Reply:
@0x009922 would you mind if we currently leave `NonZeroU64` represenation as is and left a comment about this magic schema types ticket for further discussion? 

From my point of view adding such types into the schema wouldn't add much value. Since iroha side would need to check that received value is indeed non-zero. 

So usage of `NonZeroU64` is more like safe-guard for iroha rather than distinct type.

What i'm trying to say that sending height of zero is logical error, not type error.